fn-encapsulate.js

```diff
@@ -0,0 +1,42 @@
+var path = require('path');
+var crypto = require('crypto');
+var fs = require('fs');
+
+var Module = module.constructor;
+
+module.exports = function(functionFilename, expectedFunctionName) {
```

Comment:
Not sure about "standard", but as far as I know it doesn't use internal APIs, and it is used in other places as well.

The main reason I wrote it like that is because it provides much better isolation - Similar to standard require() module loading, the impact what's being done in the module mostly remains in the module. Using eval() can lead to issues in the way that the exporter itself works (overriding methods/requires/etc.). "Hiding" the user code in a module makes this less of a problem.
- Wrapping it in a module lays the groundwork for allowing users to use actual node modules in their functions (e.g. I've actually used the crypto module in my transform function in order to generate sha256 hash codes for one of the fields during transformation). In my view, it's important to have that capability, since it empowers the transformation greatly.

On another matter, I saw in the email your comment regarding the log.info() line, and I completely agree that it shouldn't spam the command line output. However, it's being written in worker.initialize_transform() which happens only once per worker and not per hit. You're right that it needs to be written once per execution and not per worker of course.